modules/seed/addons/monitoring/monitoring.tf

```diff
@@ -148,19 +148,34 @@ module "alertmanager" {
   dns_names    = ["alertmanager.${var.ingress_base_domain}"]
 }
 
-module "alertmanager_config" {
+module "alertmanager_custom_config" {
   source = "../../../file"
   path   = "${lookup(var.config,"alertmanager_config_file","")}"
 }
 
-data "template_file" "alertmanager_config" {
-  template = "${file("${path.module}/templates/alertmanager-config.yaml")}"
+data "template_file" "alertmanager_default_config" {
+  template = "${file("${path.module}/templates/alertmanager-default-config.yaml")}"
   vars     = {}
 }
 
 locals {
-  alertmanager_default_b64 = "${base64encode(data.template_file.alertmanager_config.rendered)}"
-  alertmanager_config_b64  = "${base64encode(module.alertmanager_config.content==""?data.template_file.alertmanager_config.rendered:module.alertmanager_config.content)}"
+  alertmanager_base_config = "${module.alertmanager_custom_config.content==""?data.template_file.alertmanager_default_config.rendered:module.alertmanager_custom_config.content}"
+}
+
+data "template_file" "alertmanager_base_config_template" {
+  template = "${file("${path.module}/templates/alertmanager-base-config.yaml")}"
+
+  vars = {
+    alertmanager_smtp_host     = "${lookup(var.config,"smtp_host","smtp.example.com:587")}"
```

Comment:
I would prefer not to set defaults here.

With defaults, if the user forgets to set these values terraform will just set the default values. This will lead to a broken smtp config.

Without defaults, terraform will fail and inform the user that there are values missing. This leads the user to set sensible values instead, resulting in a working smtp config.

Reply:
Thanks @gonzolino 
You are right, Terraform will fail without defaults. But in the default case (no custom config for alertmanager provided) we even so deploy the alertmanager with the monitoring stack. The alertmanager needs a valid config to come up (which is ensured with this solution). If we had no defaults, Terraform would fail and to solve this, we are forced to provide smtp credentials, even if those are not needed in this case. In the other case (custom config provided) the user is not forced to specify real smtp config, which could lead to a scenario, where for real alerts no mails can be send out. But then it is a configuration issue.

Reply:
Ignore my previous comment :-P